lib/active_scaffold/actions/export.rb

```diff
@@ -113,6 +113,11 @@ def find_items_for_export(&block)
           :per_page => 3000,
           :page => 1
         })
+        query = beginning_of_chain.where(nil) # where(nil) is needed because we need a relation
+        # NOTE: we must use :include in the count query, because some conditions may reference other tables
+        count = count_items(query, finder_options(find_options), finder_options[:count_includes])
+        find_options[:sorting] = nil if count > 3000
```

Comment:
I'd set 3000 as a constant in this module, and then reference it everywhere.

So:
```
module ActiveScaffold::Actions
  module Export
    PAGE_SIZE = 3000
```
Then change the 2 "3000" references in this file to `PAGE_SIZE` and the "3,000" reference in `_export_form_body.html.erb` to `<%= number_with_delimiter(ActiveScaffold::Actions::Export::PAGE_SIZE) %>`